### PR TITLE
Search bar

### PR DESCRIPTION
**Pull request by heimrih**
Search bar for title

### CODE REVIEW

**Comment by heimrih** [comment]
ready

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -9,6 +9,11 @@ def index
     top = params[:top]
 
     @posts = Post.by_top(top).lastest.page(params[:page]).per Settings.posts.page.max
+
+    if params[:search]
+      @search_term = params[:search]
+      @posts = @posts.search_by(@search_term)
```
Comment (saobv-2223) [inline app/controllers/posts_controller.rb]: you can remove ()

File: app/models/post.rb [hunk 2]
```diff
@@ -10,4 +10,8 @@ class Post < ApplicationRecord
   scope :sort_by_time, ->{order created_at: :desc}
   scope :lastest, ->{order created_at: :desc}
   scope :by_top, ->(topic_id){where topic_id: topic_id if topic_id.present?}
+
+  def self.search_by(search_term)
+    where("LOWER(title) LIKE :search_term OR LOWER(user_id) LIKE :search_term", search_term: "%#{search_term.downcase}%")
+  end
```
Comment (saobv-2223) [inline app/models/post.rb]: why don't you use scope instead of this action.

**Comment by heimrih** [comment]
ready

File: app/models/post.rb [hunk 3]
```diff
@@ -10,4 +10,7 @@ class Post < ApplicationRecord
   scope :sort_by_time, ->{order created_at: :desc}
   scope :lastest, ->{order created_at: :desc}
   scope :by_top, ->(topic_id){where topic_id: topic_id if topic_id.present?}
+
+  scope :search_by, ->(search_term){
+    where "LOWER(title) LIKE :search_term OR LOWER(body) LIKE :search_term", search_term: "%#{search_term.downcase}%"}
```
Comment (saobv-2223) [inline app/models/post.rb]:
```suggestion
    scope :search_by, (lambda do |params|
      where("LOWER(title) LIKE :key OR LOWER(body) LIKE :key", key: "%#{params}%")
    end)
```
You can try to use lambda

**Comment by heimrih** [comment]
ready

**Pull request merged**